codex/contracts/interactions/clientinteractions.nim

```diff
@@ -0,0 +1,52 @@
+import pkg/ethers
+import pkg/chronicles
+import pkg/questionable
+import pkg/questionable/results
+
+import ../../purchasing
+import ../deployment
+import ../marketplace
+import ../market
+import ../proofs
+import ../clock
+import ./interactions
+
+export purchasing
+export chronicles
+
+type
+  ClientInteractions* = ref object of ContractInteractions
+    purchasing*: Purchasing
+
+proc new*(_: type ClientInteractions,
+          signer: Signer,
+          deployment: Deployment): ?!ClientInteractions =
```

Comment:
I'd avoid propagating the `Deployment` interface further into the code. We'll probably want to get rid of it (#372). Perhaps we can replace the `deployment` parameter with a `marketplaceAddress` parameter?

Reply:
Updated.